src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -0,0 +1,478 @@
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents a single EditorConfig file, see http://editorconfig.org for details about the format.
+    /// </summary>
+    public sealed partial class AnalyzerConfig
+    {
+        // Matches EditorConfig section header such as "[*.{js,py}]", see http://editorconfig.org for details
+        private static readonly Regex s_sectionMatcher = new Regex(@"^\s*\[(([^#;]|\\#|\\;)+)\]\s*([#;].*)?$", RegexOptions.Compiled);
+        // Matches EditorConfig property such as "indent_style = space", see http://editorconfig.org for details
+        private static readonly Regex s_propertyMatcher = new Regex(@"^\s*([\w\.\-_]+)\s*[=:]\s*(.*?)\s*([#;].*)?$", RegexOptions.Compiled);
+
+        /// <summary>
+        /// A set of keys that are reserved for special interpretation for the editorconfig specification.
+        /// All values corresponding to reserved keys in a (key,value) property pair are always lowercased
+        /// during parsing.
+        /// </summary>
+        /// <remarks>
+        /// This list was retrieved from https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties
+        /// at 2018-04-21 19:37:05Z. New keys may be added to this list in newer versions, but old ones will
+        /// not be removed.
+        /// </remarks>
+        public static ImmutableHashSet<string> ReservedKeys { get; }
+            = ImmutableHashSet.CreateRange(Section.PropertiesKeyComparer, new[] {
+                "root",
+                "indent_style",
+                "indent_size",
+                "tab_width",
+                "end_of_line",
+                "charset",
+                "trim_trailing_whitespace",
+                "insert_final_newline",
+            });
+
+        /// <summary>
+        /// A set of values that are reserved for special use for the editorconfig specification
+        /// and will always be lower-cased by the parser.
+        /// </summary>
+        public static ImmutableHashSet<string> ReservedValues { get; }
+            = ImmutableHashSet.CreateRange(CaseInsensitiveComparison.Comparer, new[] { "unset" });
+
+        private readonly static DiagnosticDescriptor InvalidAnalyzerConfigSeverityDescriptor
+            = new DiagnosticDescriptor(
+                "InvalidSeverityInAnalyzerConfig",
+                CodeAnalysisResources.WRN_InvalidSeverityInAnalyzerConfig_Title,
+                CodeAnalysisResources.WRN_InvalidSeverityInAnalyzerConfig,
+                "AnalyzerConfig",
+                DiagnosticSeverity.Warning,
+                isEnabledByDefault: true);
+
+        public Section GlobalSection { get; }
+
+        /// <summary>
+        /// The directory the editorconfig was contained in, with all directory separators
+        /// replaced with '/'.
+        /// </summary>
+        public string NormalizedDirectory { get; }
```

Comment:
>NormalizedDirectory [](start = 22, length = 19)

Can `NormalizedDirectory` be `internal`? It seems strange to expose a path with `\` replaced by `/` from the public API.